### ISSUE TITLE
How to catch whitespace before backslash

### ISSUE THREAD

**Issue by gamesover** [bbatsov/rubocop#5434]
Hi, I am trying to get some hints/helps from here.

Basically, I try to define my own custom rubocop rule to catch whitespace before backslash as following:
```ruby
#  bad
#  test = 'hello '\
#             'world'
#  test = 'hello '     \
#             'world'
#
#  good
#  test = 'hello ' \
#             'world'
```

So, for backslash to concatenate multilines string, it has to be one whitespace before backslash, no more, no less.
I tried to review some codes in `lib/rubocop/cop/style/line_end_concatenation.rb`, I found ruby cannot catch backslash as `token`. Yes, it can catch `+` and `<<` as tokens, but not `\`.

Would anyone gives me some directions(or example codes) how to achieve this? Thank you.

**Comment by mikegee** [comment]
This is a great idea for a cop.

I think cops are instantiated with an object describing the full source of the file to investigate. See this example:

https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/layout/empty_line_after_magic_comment.rb

Perhaps you can scan the source lines from that `source` object?

**Comment by gamesover** [comment]
@mikegee , your example is greatly helpful. I've made job done. Thanks so much for your hints!
